safekit-cluster-mirror/nestedtemplates/cluster.json

```diff
@@ -0,0 +1,650 @@
+			"copy": {
+				"name": "interfaceLoop",
+				"count": "[length(variables('vms'))]"
+			},
+			"dependsOn": [
+				"optionalLB",
+				"[resourceId('Microsoft.Network/publicIPAddresses/',concat('pip',variables('vms')[copyIndex()]))]",
+				"[resourceId('Microsoft.Network/networkSecurityGroups/',variables('NSGName'))]",
+				"[variables('virtualNetworkName')]"
+			],
+			"properties": {
+				"networkSecurityGroup": {
+					"id": "[resourceId('Microsoft.Network/networkSecurityGroups/',variables('NSGName'))]"
+				},
+				"ipConfigurations": [
+					{
+						"name": "ipconfig1",
+						"properties": {
+							"privateIPAllocationMethod": "Static",
+							"privateIPAddress": "[variables('ips')[copyIndex()]]",
+							"publicIPAddress": {
+								"id": "[resourceId('Microsoft.Network/publicIPAddresses/', concat('pip',variables('vms')[copyIndex()]))]"
+							},
+							"subnet": {
+								"id": "[resourceId('Microsoft.Network/virtualNetworks/subnets',variables('virtualNetworkName'),'Subnet')]"
+							},
+							"loadBalancerBackendAddressPools": "[variables('lbaddrpool')]"
+						}
+					}
+				]
+			}
+		},
+		{
+			"type": "Microsoft.Compute/virtualMachines",
+			"apiVersion": "2018-06-01",
+			"name": "[variables('vms')[copyIndex()]]",
+			"location": "[parameters('location')]",
+			"copy": {
+				"name": "vmLoop",
+				"count": "[length(variables('vms'))]"
+			},
+			"dependsOn": [
+				"[resourceId('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]",
+				"[resourceId('Microsoft.Network/networkInterfaces/',concat('nic',variables('vms')[copyIndex()]) )]"
+			],
+			"zones": "[split(string(add(mod(copyIndex(),3),1)), ',')]",
+			"identity": {
+				"type": "systemAssigned"
+			},
+			"properties": {
+				"hardwareProfile": {
+					"vmSize": "[parameters('vmSize')]"
+				},
+				"osProfile": {
+					"computerName": "[variables('vms')[copyIndex()]]",
+					"adminUsername": "[variables('adminUsername')]",
+					"adminPassword": "[parameters('adminPassword')]",
+					"linuxConfiguration": "[if(equals(variables('ostype'),'linux'),variables('linuxConfiguration'),'')]"
+				},
+				"storageProfile": "[if(equals(parameters('OS'),'Linux CentOS'),variables('CentosStorageProfile'),variables('windowsStorageProfile'))]",
+				"networkProfile": {
+					"networkInterfaces": [
+						{
+							"id": "[resourceId('Microsoft.Network/networkInterfaces/', concat('nic',variables('vms')[copyIndex()]))]"
+						}
+					]
+				},
+				"diagnosticsProfile": {
+					"bootDiagnostics": {
+						"enabled": true,
+						"storageUri": "[reference(resourceId(resourceGroup().name,'Microsoft.Storage/storageAccounts/', variables('storageAccountName'))).primaryEndpoints.blob]"
+					}
+				}
+			}
+		},
+		{
+			"condition": "[equals(parameters('azurePwsh'),'yes')]",
+			"type": "Microsoft.Authorization/roleAssignments",
+			"apiVersion": "2016-07-01",
+			"name": "[guid(resourceId('Microsoft.Compute/virtualMachines/',variables('vms')[copyIndex()]))]",
```

Comment:
simplify this

guid(resourceGroup().id, copyIndex())

Reply:
OK